src/FffGcodeWriter.cpp

```diff
@@ -1002,52 +1075,88 @@ void FffGcodeWriter::processInsets(GCodePlanner& gcode_layer, SliceMeshStorage*
             }
         }
-        int processed_inset_number = -1;
-        for(int inset_number=part.insets.size()-1; inset_number>-1; inset_number--)
+        // only spiralize the first part in the mesh, other parts won't be printed
+        if (spiralize && &wall_layer.parts[0] == &part)
         {
-            processed_inset_number = inset_number;
-            if (mesh->getSettingBoolean("outer_inset_first"))
+            std::vector<Polygons>& wall_insets = wall_layer.parts[0].insets;
+            if(wall_insets.size() == 0 || wall_insets[0].size() == 0)
             {
-                processed_inset_number = part.insets.size() - 1 - inset_number;
+                // wall doesn't have usable outline
+                return;
             }
-            if (processed_inset_number == 0)
+            std::vector<Polygons>* last_wall_insets = &wall_insets; // default to current wall outline
+            if (layer_nr > 0)
             {
-                if (!compensate_overlap_0)
-                {
-                    WallOverlapComputation* wall_overlap_computation(nullptr);
-                    gcode_layer.addPolygonsByOptimizer(part.insets[0], &mesh->inset0_config, wall_overlap_computation, z_seam_type, z_seam_pos, mesh->getSettingInMicrons("wall_0_wipe_dist"), spiralize);
-                }
-                else
+                SliceLayer &last_wall_layer = mesh->layers[layer_nr - 1];
+                if (last_wall_layer.parts.size() > 0 && last_wall_layer.parts[0].insets.size() > 0 && last_wall_layer.parts[0].insets[0].size() > 0)
                 {
-                    Polygons& outer_wall = part.insets[0];
-                    WallOverlapComputation wall_overlap_computation(outer_wall, mesh->getSettingInMicrons("wall_line_width_0"));
-                    gcode_layer.addPolygonsByOptimizer(outer_wall, &mesh->inset0_config, &wall_overlap_computation, z_seam_type, z_seam_pos, mesh->getSettingInMicrons("wall_0_wipe_dist"), spiralize);
+                    // use the last wall outline
+                    last_wall_insets = &last_wall_layer.parts[0].insets;
                 }
             }
-            else
+            const PolygonRef& outer_wall = wall_insets[0][0]; // current layer outer wall outline
+            const PolygonRef& last_outer_wall = (*last_wall_insets)[0][0]; // last layer outer wall outline
+            // output a wall slice that is interpolated between the last and current walls
+            const int seam_vertex_idx = wall_layer.seam_vertex_index; // current layer seam vertex index
+            const int last_seam_vertex_idx = (layer_nr == 0) ? -1 : mesh->layers[layer_nr - 1].seam_vertex_index; // last layer seam vertex index
+            gcode_layer.spiralizeWallSlice(&mesh->inset0_config, outer_wall, last_outer_wall, seam_vertex_idx, last_seam_vertex_idx);
+        }
+        else
+        {
+            int processed_inset_number = -1;
```

Comment:
Note to self: all code of this function below is *still(?)* simply the code on the left with an extra indent. No changes at all.